File: semester 3/CN/lab 1/template/clientUDP.c
// c_udp_client.c
#include <stdio.h>
#include <string.h>
#include <arpa/inet.h>
#include <unistd.h>

int main() {
    char ip[32];
    int port;

    printf("Server IP: ");
    scanf("%31s", ip);
    printf("Server port: ");
    scanf("%d", &port);

    int s = socket(AF_INET, SOCK_DGRAM, 0);

    struct sockaddr_in srv;
    memset(&srv, 0, sizeof(srv));
    srv.sin_family = AF_INET;
    srv.sin_port = htons(port);
    inet_pton(AF_INET, ip, &srv.sin_addr);

    // sending/receiving

    close(s);
    return 0;
}

File: semester 3/CN/lab 1/template/clientUDP.py
# python_udp_client.py
import socket

def main():
    ip = input("Server IP: ")
    port = int(input("Server port: "))

    s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)

    # send/recv happens in recipes file

if __name__ == "__main__":
    main()
